Improve Mac PowerPC support

ryandesign opened the pull request:
Add support for PowerPC Macs. This patch has been in the MacPorts ccache port for 6 months. See https://trac.macports.org/ticket/62516 and https://github.com/macports/macports-ports/blob/50689d1972aadbde80c9b1f60d2d8d62da7bed68/devel/ccache/files/patch-ccache-powerpc-darwin.diff.

ryandesign commented:
I'm sorry, I don't know git / GitHub well enough to understand why it decided to point this PR at master. I thought I had succeeded in making it based on dev. If you can't fix it, let me know and I can try resubmitting it.

Saalvage commented:
No worries, I rebased it for you. ^^
Only thing missing is the `NOLINT (hicpp-no-assembler)` hint and this lgtm!

ryandesign commented:
> No worries, I rebased it for you. ^^

Thanks!

> Only thing missing is the `NOLINT (hicpp-no-assembler)` hint and this lgtm!

I'm not familiar with that but I added the ` // NOLINT (hicpp-no-assembler)` comment to the end of the line to match the other lines.

Saalvage commented:
Thank you for your contribution!

barracuda156 commented:
@ryandesign We should backport this to Macports, otherwise our `doctest` still broken.
